### PR TITLE
toStringOrNull

### PR DESCRIPTION
**Pull request by DvirDukhan**
this PR adds support for `toStringOrNull` function

### CODE REVIEW

File: src/arithmetic/string_funcs/string_funcs.c [hunk 1]
```diff
@@ -412,11 +418,17 @@ void Register_StringFuncs() {
 	AR_RegFunc(func_desc);
 
 	types = array_new(SIType, 1);
-	array_append(types, SI_ALL);
+	array_append(types, STRINGABLE | T_NULL);
```
Comment (AviAvni) [inline src/arithmetic/string_funcs/string_funcs.c:421]: why this change?

Comment (swilly22) [reply src/arithmetic/string_funcs/string_funcs.c:421]: I think this is the correct change to be made
From Neo4j's doc:
```
This function will return an error if provided with an expression that is not an integer, float, string, boolean, point, duration, date, time, localtime, localdatetime or datetime value.
```

File: src/arithmetic/string_funcs/string_funcs.c [hunk 2]
```diff
@@ -13,6 +13,8 @@
 #include "../../util/strutil.h"
 #include "../../util/json_encoder.h"
 
+#define STRINGABLE (SI_NUMERIC | T_POINT | T_DURATION | T_DATETIME| T_STRING | T_BOOL)
```
Comment (swilly22) [inline src/arithmetic/string_funcs/string_funcs.c:16]: Please add a one line comment explaining the purpose of this define
